gateway/webhook/setup.go

```diff
@@ -18,25 +21,35 @@ import (
 	"github.com/rudderlabs/rudder-go-kit/logger"
 	"github.com/rudderlabs/rudder-go-kit/stats"
 
-	gwstats "github.com/rudderlabs/rudder-server/gateway/internal/stats"
-	gwtypes "github.com/rudderlabs/rudder-server/gateway/internal/types"
-	"github.com/rudderlabs/rudder-server/gateway/webhook/model"
-	"github.com/rudderlabs/rudder-server/services/transformer"
 	"github.com/rudderlabs/rudder-server/utils/crash"
 )
 
 type Gateway interface {
-	TrackRequestMetrics(errorMessage string)
+	RequestMetricsTracker
+	WebhookHandler
+}
+
+type WebhookHandler interface {
 	ProcessWebRequest(writer *http.ResponseWriter, req *http.Request, reqType string, requestPayload []byte, arctx *gwtypes.AuthRequestContext) string
-	NewSourceStat(arctx *gwtypes.AuthRequestContext, reqType string) *gwstats.SourceStat
+	NewSourceStatReporter(arctx *gwtypes.AuthRequestContext, reqType string) gwtypes.StatReporter
 	SaveWebhookFailures([]*model.FailedWebhookPayload) error
 }
 
+// RequestMetricsTracker is used to track webhook request metrics on a request basis for OSS customers
+type RequestMetricsTracker interface {
+	TrackRequestMetrics(errorMessage string)
+}
+
 type WebHookI interface {
 	RequestHandler(w http.ResponseWriter, r *http.Request)
 	Register(name string)
 }
 
+type TransformerFeaturesService interface {
+	SourceTransformerVersion() string
+	Wait() chan struct{}
+}
```

Comment:
Can't we simplify this one like we did in the ingestion-svc and just have a function like `type TransformerFeaturesVersionFetcher func () string`?

We either make sure that the webhook doesn't receive traffic is the featuresSvc isn't ready (like we do in ingestion-svc) or we make sure that the function itself blocks until ready first and then returns the version. I think the former would be preferrable. Wdyt?